arygm/QuickFix
# feature: switch worker button is responsive

# **Add Worker Mode Switching on Welcome Screen**

## **Description**
This PR introduces a new feature that allows users to switch to worker mode by clicking the **"Switch to Worker"** button on the Welcome Onboard screen at the end of the registration process as a worker.

### **Key Features**
- **Seamless Worker Mode Activation:**
  - Users can now activate worker mode directly from the Welcome Onboard screen.
  - The feature is triggered by clicking the **"Switch to Worker"** button.
- **Enhanced User Experience:**
  - Provides a smooth transition from user mode to worker mode immediately after registration as a worker.

### **Implementation Details**
- Added a new button labeled **"Switch to Worker"** on the Welcome Onboard screen.
- Implemented the logic to:
  - Update the user account to worker mode in the backend.
  - Transition the UI to worker-specific features and interfaces.
- Integrated state management to handle the switch seamlessly.

### **Testing**
- Verified that clicking the **"Switch to Worker"** button successfully transitions to worker mode.
- Ensured the backend updates the user's account status to worker.
- Tested UI responsiveness and stability during the mode switch.

### **Screenshots**
*Add screenshots or GIFs of the feature in action if available.*

---

## **Related Issues**
*Link any related issues here, if applicable.*

---

## **Steps to Test**
1. Complete the registration process as a worker.
2. On the Welcome Onboard screen, click the **"Switch to Worker"** button.
3. Verify that:
   - The app transitions to worker mode.
   - Worker-specific features and interfaces are enabled.

---

### **Notes**
- The UI/UX for worker-specific functionality will be enhanced in subsequent PRs.
- Further optimizations for the backend logic will follow in future updates.
